raft-tech/TANF-app
# User viewset not returning/duplicating users

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2551 

## How to Test
- ON THE `develop` BRANCH:
```
cd tdrs-frontend && docker-compose up 
cd tdrs-backend && docker-compose up
```
- Apply the attached patch `users.patch`
- ```  cd tdrs-backend && docker-compose exec web pythons manage.py generate_cypress_users ```
- Log into frontend/backend as admin
- Go to localhost:3000/v1/users and check to see if users have been duplicated on the last page of the view. Note, it doesnt always happen because of the way Django creates the DB query. You may have to teardown and retry generating all the users a few times to see this issue arise. 
- Now perform the same steps as above with the `2511-users-viewset` branch

[users.patch](https://github.com/raft-tech/TANF-app/files/11728059/users.patch)

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] All users are returned 
+ [x] Pagination works on the local environment

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
